src/components/tooltip/tooltip.js

```diff
@@ -3,392 +3,396 @@
-      if (element.text().indexOf($interpolate.startSymbol()) > -1) {
-        scope.$watch(function() {
-          return element.text().trim();
-        }, addAriaLabel);
+    function setDefaults() {
+      scope.mdDelay = scope.mdDelay || TOOLTIP_SHOW_DELAY;
+      if (!DIRECTIONS[scope.mdDirection]) {
+        scope.mdDirection = TOOLTIP_DIRECTION;
       }
     }
 
-    function addAriaLabel (override) {
-      if ((override || !parent.attr('aria-label')) && !parent.text().trim()) {
+    function addAriaLabel(override) {
+      if (override || !parent.attr('aria-label')) {
         var rawText = override || element.text().trim();
         var interpolatedText = $interpolate(rawText)(parent.scope());
         parent.attr('aria-label', interpolatedText);
       }
     }
 
-    function manipulateElement () {
-      element.detach();
-      element.attr('role', 'tooltip');
-    }
+    function updatePosition() {
+      setDefaults();
 
-    function bindEvents () {
-      var mouseActive = false;
+      // If the panel has already been created, remove the current origin
+      // class from the panel element.
+      if (panelRef && panelRef.panelEl) {
+        panelRef.panelEl.removeClass(origin);
+      }
 
-      // add an mutationObserver when there is support for it
-      // and the need for it in the form of viable host(parent[0])
+      // Set the panel element origin class based off of the current
+      // mdDirection.
+      origin = 'md-origin-' + scope.mdDirection;
+
+      // Create the position of the panel based off of the mdDirection.
+      position = DIRECTIONS[scope.mdDirection];
+
+      // Using the newly created position object, use the MdPanel
+      // panelPosition API to build the panel's position.
+      panelPosition = $mdPanel.newPanelPosition()
+          .relativeTo(parent)
+          .addPanelPosition(
+            $mdPanel.xPosition[position.x],
+            $mdPanel.yPosition[position.y]
+          );
+
+      // If the panel has already been created, add the new origin class to
+      // the panel element and update it's position with the panelPosition.
+      if (panelRef && panelRef.panelEl) {
+        panelRef.panelEl.addClass(origin);
+        panelRef.updatePosition(panelPosition);
+      }
+    }
+
+    function bindEvents() {
+      // Add a mutationObserver where there is support for it and the need
+      // for it in the form of viable host(parent[0]).
       if (parent[0] && 'MutationObserver' in $window) {
-        // use an mutationObserver to tackle #2602
+        // Use a mutationObserver to tackle #2602.
         var attributeObserver = new MutationObserver(function(mutations) {
-          if (mutations.some(function (mutation) {
-              return (mutation.attributeName === 'disabled' && parent[0].disabled);
-            })) {
-              $mdUtil.nextTick(function() {
-                setVisible(false);
-              });
+          if (mutations.some(function(mutation) {
```

Comment:
This is a tad confusing in an if statement. Consider making a function (isDisabled or something) to check this.